torch/_dynamo/variables/builder.py

```diff
@@ -1273,7 +1273,9 @@ def wrap_tensor(self, value: torch.Tensor):
         ):
             unimplemented("torch.compile does not support strided NestedTensor")
 
-        if is_sparse_any(value):
+        # Reject sparse, but not coo.
+        # TODO: remove this altogether when non-coo sparsity propagation is ready
+        if is_sparse_any(value) and not value.is_sparse:
```

Comment:
This change triggers the `TestTorchTidyProfiler::test_sparse_tensors` failure: in the main branch, a graph break is triggered when `value` is sparse coo tensor which apparently avoids the failure for the main.

Reply:
The following hack fixes the `TestTorchTidyProfiler::test_sparse_tensors` failure:
```
diff --git a/torch/_dynamo/variables/torch.py b/torch/_dynamo/variables/torch.py
index 8e7089f0805..12533727e13 100644
--- a/torch/_dynamo/variables/torch.py
+++ b/torch/_dynamo/variables/torch.py
@@ -249,7 +249,7 @@ class TorchCtxManagerClassVariable(BaseTorchVariable):
             torch.cpu.amp.autocast,
         ):
             return AutocastModeVariable.create(self.value, args, kwargs)
-        elif self.value in (
+        elif 0 and self.value in (
             torch.profiler.profile,
             torch.profiler.record_function,
             torch.autograd.profiler.profile,
```

@ezyang do you have ideas how to resolve this issue properly? Skipping `profile` by just returning None above seems to be the cause of having `test/dynamo_expected_failures/TestTorchTidyProfiler.test_*` in the first place.

Reply:
https://github.com/pytorch/pytorch/pull/126965 - let's see if this works..

Reply:
ok, it didn't work and there seems to be no easy fix to make profile work when dynamo is enabled.

@aartbik I suggest the following fix: create empty file
```
test/dynamo_expected_failures/TestTorchTidyProfiler.test_sparse_tensors
```
and add it to your PR.